dev/com.ibm.ws.el.3.0_fat/fat/src/com/ibm/ws/el/fat/tests/EL30MiscTests.java

```diff
@@ -0,0 +1,140 @@
+        WebConversation wc = new WebConversation();
+
+        String url = ELUtils.createHttpUrlString(elServer, ServiceLookup_AppName, "TestServiceLookup");
+        LOG.info("url: " + url);
+
+        WebRequest request = new GetMethodWebRequest(url);
+        WebResponse response = wc.getResponse(request);
+        LOG.info("Servlet response : " + response.getText());
+
+        assertEquals("Expected " + 200 + " status code was not returned!",
+                     200, response.getResponseCode());
+        assertTrue("The response did not contain: Lookup works!", response.getText().contains("Lookup works!"));
+    }
+
+    /**
+     * Comments in the META-INF/services/javax.el.ExpressionFactory files should be ignored
+     *
+     * The BZ 64097 patch was applied to both EL-3.0 and th EL-2.2 API.
+     *
+     * This one tests the service look using the jsp-2.2 feature because it relies on the el 2.2 api. This test was placed here for two mains reasons:
+     * - There's no el 2.2 bucket.
+     * - This scenario tests EL, so we chose not to place this in the the jsp FATs.
+     * - We didn't want to duplicate the test application in multiple buckets
+     *
+     * https://github.com/OpenLiberty/open-liberty/pull/18424
+     *
+     * @throws Exception
+     */
+    @SkipForRepeat(SkipForRepeat.EE9_FEATURES)
+    @Test
+    public void testEL22ServiceLookup() throws Exception {
+
+        elServer.saveServerConfiguration();
+
+        ServerConfiguration configuration = elServer.getServerConfiguration();
+        LOG.info("Server configuration that was saved: " + configuration);
+        elServer.setMarkToEndOfLog();
+        elServer.setServerConfigurationFile("serverConfigs/jsp22server.xml");
+        elServer.waitForConfigUpdateInLogUsingMark(Collections.singleton(ServiceLookup_AppName), true, "CWWKT0016I:.*ELServicesLookup.*");
+
+        configuration = elServer.getServerConfiguration();
+        LOG.info("Updated server configuration: " + configuration);
+
+        WebConversation wc = new WebConversation();
+
+        wc.setExceptionsThrownOnErrorStatus(false);
+
+        String url = ELUtils.createHttpUrlString(elServer, ServiceLookup_AppName, "EL22Test.jsp");
+        LOG.info("url: " + url);
+
+        WebRequest request = new GetMethodWebRequest(url);
+        WebResponse response = wc.getResponse(request);
+        LOG.info("Servlet response : " + response.getText());
+
+        assertEquals("Expected " + 200 + " status code was not returned!",
+                     200, response.getResponseCode());
+        assertTrue("The response did not contain: Lookup works!", response.getText().contains("Lookup works!"));
+
+        elServer.setMarkToEndOfLog();
+        elServer.restoreServerConfiguration();
+        // Wait for the application that is still deployed to start.
+        elServer.waitForConfigUpdateInLogUsingMark(Collections.singleton(ServiceLookup_AppName), true, "CWWKT0016I:.*ELServicesLookup.*");
```

Comment:
Can we do this in a try-finally to ensure if there is an exception from one of the asserts that we actually restore the server.xml back to the original configuration?

Reply:
Let me know how it looks now